cas-server-support-generic/src/main/java/org/jasig/cas/adaptors/generic/ShiroAuthenticationHandler.java

```diff
@@ -36,11 +41,32 @@
  * @author Misagh Moayyed
  * @since 4.2
  */
+@Component("shiroAuthenticationHandler")
 public class ShiroAuthenticationHandler extends AbstractUsernamePasswordAuthenticationHandler {
 
     private Set<String> requiredRoles = new HashSet<>();
     private Set<String> requiredPermissions = new HashSet<>();
 
+    @Value("${shiro.authn.requiredRoles:}")
+    private String requiredRolesConfig;
+
+    @Value("${shiro.authn.requiredPermissions:}")
+    private String requiredPermissionsConfig;
+
+    /**
+     * Initialize roles and permissions.
+     */
+    @PostConstruct
+    public void init() {
+        if (StringUtils.isNotBlank(this.requiredPermissionsConfig)) {
+            setRequiredPermissions(org.springframework.util.StringUtils.commaDelimitedListToSet(this.requiredPermissionsConfig));
```

Comment:
Yup. It's very helpful when converting properties to collections. I resorted to doing this conversion via Java, because writing that line in Spring EL was just very confusing, and likely not possible.